test/CMakeLists.txt

```diff
@@ -16,12 +17,12 @@
 # want to import these projects into their IDE.
 #
 cmake_minimum_required(VERSION 3.5)
-  
-add_subdirectory(.. ${CMAKE_CURRENT_BINARY_DIR}/boost_config)
-  
-add_executable(config_info config_info.cpp)  
-target_link_libraries(config_info Boost::config)  
 
-add_executable(config_test config_test.cpp)  
-target_link_libraries(config_test Boost::config)  
+find_package(Threads)
```

Comment:
Why does this cmake self test have to enable multithreading?

Reply:
The test requires it (fails with pthread error if omitted):

```
[ 25%] Linking CXX executable config_test
/usr/bin/ld: CMakeFiles/config_test.dir/config_test.cpp.o: in function `boost_has_pthread_mutexattr_settype::f()':
config_test.cpp:(.text+0x5ace): undefined reference to `pthread_mutexattr_init'
/usr/bin/ld: config_test.cpp:(.text+0x5ae6): undefined reference to `pthread_mutexattr_settype'
/usr/bin/ld: CMakeFiles/config_test.dir/config_test.cpp.o: in function `boost_has_pthread_yield::f()':
config_test.cpp:(.text+0x5b15): undefined reference to `pthread_yield'
/usr/bin/ld: CMakeFiles/config_test.dir/config_test.cpp.o: in function `boost_has_pthreads::test()':
config_test.cpp:(.text+0x5b93): undefined reference to `pthread_mutex_trylock'
/usr/bin/ld: config_test.cpp:(.text+0x5bd1): undefined reference to `pthread_create'
/usr/bin/ld: config_test.cpp:(.text+0x5bed): undefined reference to `pthread_join'
collect2: error: ld returned 1 exit status
make[2]: *** [test/CMakeFiles/config_test.dir/build.make:84: test/config_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:93: test/CMakeFiles/config_test.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```